packages/client/src/components/app/filter/FilterButton.svelte

```diff
@@ -0,0 +1,250 @@
+
+      // Process as single if the operator requires it
+      if (!isArrayOperator(filter.operator)) {
+        const userRow = $rowCache?.[filter.value]
+        userDisplay = userRow?.email ?? filter.value
+      } else {
+        const filterVals = Array.isArray(filter.value)
+          ? filter.value
+          : [filter.value]
+
+        // Email is currently the default display field for users.
+        userDisplay = parseMultiDisplay(
+          filterVals.map((val: string) => $rowCache?.[val]?.email ?? val)
+        )
+      }
+
+      display = userDisplay
+    }
+
+    return `${filterOp?.label.toLowerCase()} ${filter.noValue ? "" : display}`
+  }
+</script>
+
+<FilterPopover
+  bind:this={popover}
+  {filter}
+  {operators}
+  {schema}
+  {config}
+  on:change
+>
+  <div slot="anchor">
+    <div class="filter-button-wrap" class:inactive={!filter}>
+      <!-- svelte-ignore a11y-click-events-have-key-events -->
+      <!-- svelte-ignore a11y-no-static-element-interactions -->
+      <div
+        bind:this={button}
+        class:is-disabled={disabled}
+        class="new-styles spectrum-Button spectrum-Button--secondary spectrum-Button--size{size.toUpperCase()}"
+        title={filterTitle || filterDisplay}
+      >
+        <div
+          class="toggle-wrap"
+          on:click={e => {
+            if (filter) {
+              e.stopPropagation()
+            }
+            if (!disabled) {
+              dispatch("toggle")
+            }
+          }}
+        >
+          <svg
+            class="spectrum-Icon spectrum-Icon--sizeS"
+            focusable="false"
+            aria-hidden="true"
+            aria-label={icon}
+          >
+            <use xlink:href="#spectrum-icon-18-{icon}" />
+          </svg>
+        </div>
+
+        <span class="spectrum-Button-label" class:truncate>
+          <span class="field">{config?.label || config?.field}</span>
+          {#if filter}
+            <span class="display">
+              {filterDisplay}
+            </span>
+          {/if}
+        </span>
+        {#if filterMeta}
+          <span class="filterMeta">{filterMeta}</span>
+        {/if}
+      </div>
+    </div>
+  </div>
+</FilterPopover>
+
+<style>
+  .toggle-wrap {
+    z-index: 1;
+  }
+  .toggle-wrap svg {
+    width: 12px;
+    pointer-events: none;
+  }
+  .filter-button-wrap.inactive .spectrum-Button .spectrum-Icon {
+    color: var(--grey-6);
+  }
+
+  .spectrum-Button {
+    position: relative;
+    display: flex;
+    gap: var(--spacing-xs);
+  }
+  .spectrum-Button.is-disabled {
+    cursor: default;
+  }
+
+  .filter-button-wrap.inactive .spectrum-Button {
+    background: var(--spectrum-global-color-gray-50);
+    border: 1px dashed var(--spectrum-global-color-gray-200);
+  }
+
+  .spectrum-Button-label.truncate {
+    white-space: nowrap;
+    overflow: hidden;
+    text-overflow: ellipsis;
+    max-width: 200px;
+  }
+
+  .spectrum-Button-label .display {
+    color: rgba(179, 222, 254, 1);
```

Comment:
I find it quite hard to read the filters when applied:
![image](https://github.com/user-attachments/assets/44986de4-d584-44a8-9d61-3a8cf3013e2c)
We should probably use a theme aware variable with more contrast 👍